### PR TITLE
cut hard dependency on requests

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/__init__.py [hunk 1]
```diff
@@ -26,82 +26,131 @@
...
+    from ._requests_basic import RequestsTransport, RequestsTransportResponse
     __all__.extend([
-        'AsyncHttpTransport',
-        'AsyncHttpResponse',
-        'AsyncioRequestsTransport',
-        'AsyncioRequestsTransportResponse'
+        'RequestsTransport',
+        'RequestsTransportResponse',
     ])
+    try:
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/transport/__init__.py:43]: Could we add comments to all the the try/except blocks here? The logic is a little hard to follow :)

File: sdk/core/azure-core/azure/core/utils/_utils.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def _case_insensitive_dict(*args, **kwargs):
         # multidict is installed by aiohttp
         from multidict import CIMultiDict
 
+        if len(kwargs) == 0 and len(args) == 1 and (not args[0]):
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/utils/_utils.py:102]: What's `args[0]` in this case? Maybe add a small comment as to what this if-statement is addressing?

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/utils/_utils.py:102]: This is to handle case_insensitive_dict(None). requests.structures.CaseInsensitiveDict returns empty dict while multidict.CIMultiDict raise error.

Here we want to return dict().

**Review by annatisch — APPROVED** [review]
